cmd/builder/internal/config/default.go

```diff
@@ -0,0 +1,32 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"embed"
+
+	"github.com/knadh/koanf"
+	"github.com/knadh/koanf/providers/fs"
+)
+
+//go:embed *.yaml
+var configs embed.FS
+
+// DefaultProvider returns a koanf.Provider that provides the default build
+// configuration file. This is the same configuration that otelcorecol is
+// built with.
+func DefaultProvider() koanf.Provider {
+	return fs.Provider(configs, "default.yaml")
```

Comment:
Do we need to use FS here or could we use file.Provider instead?  I ask bc Fs is listed as experimental still.

Reply:
Where is it marked experimental?

https://pkg.go.dev/github.com/knadh/koanf@v1.4.2/providers/fs

Reply:
I found it marked experimental on the github page: https://github.com/knadh/koanf#bundled-providers.  That particular line hasn't been updated in 13 months, so could be out of date.

Reply:
I see. I was looking at the [source](https://github.com/knadh/koanf/blob/v1.4.2/providers/fs/fs.go) and godoc, which doesn't mention experimental status. We can switch to the file provider, but the code needed for that will be basically the same as the fs one, so I don't think there's much benefit.